### ISSUE TITLE
Log Analytics: Difference between TimeGenerated and _timestamp_d

### ISSUE THREAD

**Issue by XavierGeerinck** [microsoft/azure-container-apps#79]
When working with Container Apps, it seems almost impossible to make sense of logs. When querying the `ContainerAppConsoleLogs_CL` table it returns `TimeGenerated` and `_timestamp_d`, however when ordering they do not result in the same ordering.

This might be due to that `TimeGenerated` ["contains the date and time that the record was created by the data source."](https://docs.microsoft.com/en-us/azure/azure-monitor/logs/log-standard-columns) but this makes me wonder: why does the data source generate a log this late? 

The other question that remains is: what is `_timestamp_d` then? Is this the actual log file?

Here is some log output that I got, note the 7:58 -> 8:03 -> 7:58 -> 8:00 weird ordering on `TimeGenerated`

|TimeGenerated|Container App Name|_timestamp_d|
|-|-|-|
| 1/18/2022, 7:58:50.306 AM | app6e4b4997 | 1,642,492,729.125 |  
| 1/18/2022, 7:58:45.375 AM | app6e4b4997 | 1,642,492,724.124 |  
| 1/18/2022, 8:03:06.632 AM | app6e4b4997 | 1,642,492,724.083 |  
| 1/18/2022, 8:03:06.632 AM | app6e4b4997 | 1,642,492,724.083 |  
| 1/18/2022, 8:03:06.632 AM | app6e4b4997 | 1,642,492,724.083 |  
| 1/18/2022, 8:03:06.632 AM | app6e4b4997 | 1,642,492,724.083 |  
| 1/18/2022, 8:03:06.632 AM | app6e4b4997 | 1,642,492,724.083 |  
| 1/18/2022, 8:03:06.632 AM | app6e4b4997 | 1,642,492,724.083 |  
| 1/18/2022, 8:03:06.632 AM | app6e4b4997 | 1,642,492,723.982 |  
| 1/18/2022, 8:03:06.632 AM | app6e4b4997 | 1,642,492,723.982 |  
| 1/18/2022, 7:58:44.511 AM | app6e4b4997 | 1,642,492,723.912 |  
| 1/18/2022, 7:58:44.511 AM | app6e4b4997 | 1,642,492,723.911 |  
| 1/18/2022, 8:03:58.771 AM | app6e4b4997 | 1,642,492,723.910 |  
| 1/18/2022, 8:01:45.374 AM | app6e4b4997 | 1,642,492,646.119 |  
| 1/18/2022, 8:01:45.374 AM | app6e4b4997 | 1,642,492,646.119 |  
| 1/18/2022, 8:02:54.297 AM | app6e4b4997 | 1,642,492,645.360 |  
| 1/18/2022, 8:00:05.551 AM | app6e4b4997 | 1,642,492,476.853 |  
| 1/18/2022, 8:00:00.865 AM | app6e4b4997 | 1,642,492,471.852 |  
| 1/18/2022, 7:59:28.549 AM | app6e4b4997 | 1,642,492,471.362 |  
| 1/18/2022, 7:59:28.549 AM | app6e4b4997 | 1,642,492,471.362 |  
| 1/18/2022, 7:59:28.549 AM | app6e4b4997 | 1,642,492,471.362 |

In summary, I have 2 questions:

- [ ] What does `_timestamp_d` represent?
- [ ] Why is `TimeGenerated` so much different compared to `_timestamp_d`?

**Comment by sanchitmehta** [comment]
`_timestamp_d` represents the timestamp when Container Runtime records the log line. `TimeGenerated` is the log line when the log agent running in the environment records the log line. The agent processes log lines in batches and can have a delay in  proccessing the log entry. Ideally, this delay is few seconds. Could you provide us with the environment name + resource group? We can investigate the latencies

**Comment by XavierGeerinck** [comment]
Hi @sanchitmehta this is with each container app, I make one, it logs ~20 lines and TimeGenerated will always have an unordered list.

Right now I run the following which works:

```
ContainerAppConsoleLogs_CL 
    | project TimeCreated = unixtime_milliseconds_todatetime(_timestamp_d * 1000), ContainerAppName_s, Log_s
    | order by TimeCreated desc 
    | take 150
```

**Comment by SophCarp** [comment]
I see you have an answer, I will resolve this issue. Additionally, I'm making a note to update the docs with logging details. Thank you! Please reopen if you need additional information.
